crates/bevy_gizmos/src/primitives/dim2.rs

```diff
@@ -0,0 +1,434 @@
+//! A module for rendering each of the 2D [`bevy_math::primitives`] with [`Gizmos`].
+
+use super::helpers::*;
+
+use bevy_math::primitives::{
+    BoxedPolygon, BoxedPolyline2d, Circle, Direction2d, Ellipse, Line2d, Plane2d, Polygon,
+    Polyline2d, Primitive2d, Rectangle, RegularPolygon, Segment2d, Triangle2d,
+};
+use bevy_math::{Mat2, Vec2};
+use bevy_render::color::Color;
+
+use crate::prelude::{GizmoConfigGroup, Gizmos};
+
+/// A trait for rendering 2D geometric primitives (`P`) with [`Gizmos`].
+pub trait GizmoPrimitive2d<P: Primitive2d> {
+    /// The output of `primitive_2d`. This is a builder to set non-default values.
+    type Output<'a>
+    where
+        Self: 'a;
+
+    /// Renders a 2D primitive with its associated details.
+    fn primitive_2d(
+        &mut self,
+        primitive: P,
+        position: Vec2,
+        rotation: Mat2,
+        color: Color,
+    ) -> Self::Output<'_>;
+}
+
+// direction 2d
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive2d<Direction2d> for Gizmos<'w, 's, T> {
+    type Output<'a> = () where Self : 'a;
+
+    fn primitive_2d(
+        &mut self,
+        primitive: Direction2d,
+        position: Vec2,
+        rotation: Mat2,
+        color: Color,
+    ) -> Self::Output<'_> {
+        if !self.enabled {
+            return;
+        }
+
+        let start = position;
+        let end = position + (rotation * *primitive);
+        self.arrow_2d(start, end, color);
```

Comment:
This will currently draw an arrow with a length of 1 pixel with a default 2D camera. Not entirely sure what the default value *should* be, maybe 50 pixels? It looks decent, but it's quite arbitrary... Either way, the arrow length should probably be configurable.